src/mlpack/methods/dbscan/dbscan_impl.hpp

```diff
@@ -164,53 +184,62 @@ void DBSCAN<RangeSearchType, PointSelectionPolicy>::PointwiseCluster(
   // We maintain a list of non-core points so that we can handle that logic
   // correctly.
 
-  std::vector<bool> visited(data.n_cols, false);
-  std::vector<bool> nonCorePoints(data.n_cols, false);
+  std::vector<std::atomic<bool>> visited(data.n_cols);
+  std::vector<std::atomic<bool>> nonCorePoints(data.n_cols);
 
   for (size_t i = 0; i < data.n_cols; ++i)
   {
-    if (i % 10000 == 0 && i > 0)
-      Log::Info << "DBSCAN clustering on point " << i << "..." << std::endl;
-
-    // Get the next index.
-    const size_t index = pointSelector.Select(i, data);
```

Comment:
The use of `pointSelector` to allow custom next point selection strategies is also important and we shouldn't remove it.